torch/optim/adamw.py

```diff
@@ -396,8 +398,13 @@ def _single_tensor_adamw(
 
             # 1 - beta1 ** step can't be captured in a CUDA graph, even if step is a CUDA tensor
             # (incurs "RuntimeError: CUDA error: operation not permitted when stream is capturing")
-            bias_correction1 = 1 - torch.pow(beta1, step)
-            bias_correction2 = 1 - torch.pow(beta2, step)
+            if _is_xla(step):
+                # `float_power` is more accurate when using xla bf16.
+                bias_correction1 = 1 - torch.float_power(beta1, step)
```

Comment:
This looks very surprising as it is a semantic change to what the operator does. This means that the correction will always be computed in float64 on xla device? Note that this will also impact all the other ops below and turn all of them in float64 ops!

Note that the non-capturable flow is not using float_power but `**` and so will preserve the dtype of the output contrary to this change.

Could you give more details why pow is different from `**` for you?

Reply:
Sure! It is *not* the case that `float_power` computation will be done in float64 on the xla device. Internally, it will be lowered to an fp32 computation. 

`%power = f32[] power(f32[] %p1, f32[] %convert), metadata={op_type="aten__pow" op_name="aten__pow"}`

Results for the different ops:
`pow - 1.`
`float_power - 0.994`
`** - 0.994`

Test code for reference:
```
import torch
import torch_xla.core.xla_model as xm

beta = 0.995
def func1(step):
    step = torch.tensor(step, device=xm.xla_device())
    factor = 1 - torch.pow(beta, step)
    print(factor)

def func2(step):
    step = torch.tensor(step, device=xm.xla_device())
    factor = 1 - torch.float_power(beta, step)
    print(factor)

def func3(step):
    factor = 1 - (beta ** step)
    print(factor)

func1(100)
func2(100)
func3(100)
```

Reply:
Interesting!
I'm wondering if we could move this back to just be `1 - beta1 ** step` in all cases then. Can you try to see if that passes CI please?

Reply:
On further reflection `pow` would cause all optimizer computations to be in bf16 when using `XLA_USE_BF16` or `XLA_DOWNCAST_BF16`. Maybe `float_power` is the way to go.

Pinging @wconstab and @JackCaoG for comments.